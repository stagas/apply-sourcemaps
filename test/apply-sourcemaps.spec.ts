/** @env node */
import * as path from 'path'
import { puppito } from 'puppito'
import { applySourceMaps, clearAllCaches } from '../src/index.ts'

const homedir = path.resolve(__dirname, '..')

const getContext = (inlineSourceMaps: boolean) => {
  const ctx: any = {
    logs: [],
  }

  beforeAll(async () => {
    const p = ctx.p = await puppito({
      homedir,
      file: path.join(__dirname, 'fixtures', 'a', 'index.ts'),
      quiet: true,
      silent: true,
      inlineSourceMaps,
      transformArgs: async (args, originUrl) => {
        ctx.logs.push([originUrl, args])
        return args
      },
      failedRequestFilter: x => {
        return !x.includes('/onreload') && !x.includes('.css')
      },
    })
    await p.page.goto(ctx.p.server.url)
  })

  afterAll(async () => {
    await ctx.p.close()
    clearAllCaches()
  })

  afterEach(() => {
    ctx.logs = []
  })

  ctx.run = async () => {
    await ctx.p.flush()

    const urls: string[] = []

    const out = await Promise.all(ctx.logs.map(
      async ([, args]: any) => {
        for (const [i, x] of args.entries()) {
          if (typeof x !== 'string') continue
          args[i] = await applySourceMaps(args[i], url => {
            urls.push(url)
            return url
          })
        }
        return args
      }
    ))

    return { urls, out }
  }
  return ctx
}

describe('applySourceMaps', () => {
  describe('http', () => {
    describe('external', () => {
      const ctx = getContext(false)

      it('should work', async () => {
        const { out, urls } = await ctx.run()
        expect(out).toMatchSnapshot()
        expect(urls).toMatchSnapshot()
      })
    })

    describe('inline', () => {
      const ctx = getContext(true)

      it('should work', async () => {
        const { out, urls } = await ctx.run()
        expect(out).toMatchSnapshot()
        expect(urls).toMatchSnapshot()
      })
    })
  })
})
