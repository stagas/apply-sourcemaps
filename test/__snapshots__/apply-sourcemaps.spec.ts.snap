
exports[`applySourceMaps http external should work 1`] = `
Array [
  Array [
    "onetwo",
  ],
  Array [
    "/@fs/test/fixtures/a/index.ts:8:9",
  ],
]
`;

exports[`applySourceMaps http external should work 2`] = `
Array [
  "/@fs/test/fixtures/a/index.ts",
]
`;

exports[`applySourceMaps http inline should work 1`] = `
Array [
  Array [
    "onetwo",
  ],
  Array [
    "/@fs/test/fixtures/a/index.ts:8:9",
  ],
]
`;

exports[`applySourceMaps http inline should work 2`] = `
Array [
  "/@fs/test/fixtures/a/index.ts",
]
`;
